x/ethbridge/handler.go

```diff
@@ -8,16 +8,19 @@ import (
 
 	"github.com/cosmos/cosmos-sdk/codec"
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	"github.com/cosmos/cosmos-sdk/x/bank"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	"github.com/cosmos/cosmos-sdk/x/supply"
 )
 
 // NewHandler returns a handler for "ethbridge" type messages.
-func NewHandler(oracleKeeper oracle.Keeper, bankKeeper bank.Keeper, codespace sdk.CodespaceType, cdc *codec.Codec) sdk.Handler {
+func NewHandler(oracleKeeper oracle.Keeper, supplyKeeper supply.Keeper, accountKeeper auth.AccountKeeper, codespace sdk.CodespaceType, cdc *codec.Codec) sdk.Handler {
```

Comment:
I think it's time to start using `expected_keepers`, which are interfaces that _only_ expose the necessary functions from the keepers

Reply:
also why are we passing a codespace directly to the Handler instead of using the keeper's one?

Reply:
will refactor to use expected_keepers and to create a bridge keeper in follow up PR